test/karma/karma-failed-tests-reporter.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/* 
+ * Credits:
+ * This karma reporter was written by Pascal Birchler:
+ *
+ * https://github.com/GoogleForCreators/web-stories-wp/pull/4676
+ * https://github.com/swissspidy
+ */
```

Comment:
this was written by @swissspidy (thank you!!):
https://github.com/GoogleForCreators/web-stories-wp/pull/4676

I spent some time trying to get a solution with jasmine but ended up re-using Pascal's work because:
1. it works for our use case (re-running failures on browser tests)
2. karma is deprecated, we'll probably move to web-test-runner or other option in a few months:

https://github.com/karma-runner/karma#karma-is-deprecated-and-is-not-accepting-new-features-or-general-bug-fixes

jasmine doesn't support retries:
https://github.com/jasmine/jasmine/issues/960#issuecomment-160749271